flowmachine/flowmachine/core/server/server.py

```diff
@@ -79,6 +79,10 @@ def get_reply_for_message(msg_str: str) -> ZMQReply:
         return ZMQReply(
             status="error", msg="Invalid JSON.", payload={"decode_error": exc.msg}
         )
+    except Exception as exc:
+        # Generic error boundary to catch any unexpected errors and pass to API
+        return ZMQReply(status="error", msg=f"{exc}")
+        raise exc
```

Comment:
This line, of course, is pointless since it happens after `return`. Need to rethink this a bit.

Reply:
Hmm. Well, obvs just remove the unreachable return?

I think my issue with this is you don't really want to be returning uncensored exceptions from a pokeable service, right? Plus, we'd want to be able disambiguate a 400 (you screwed up) from a 500 (we screwed up) so we can return a correct status code from the API.

Reply:
Yes, perhaps we just want to send back a "something went wrong" message in this case, rather than the actual error message. My aim is to make sure we return _something_,  because at the moment if FlowMachine raises an error that the server isn't expecting it will just leave the API hanging, so from the user's perspective it's not obvious what's happened.

Reply:
As for the `raise` after `return`, I was thinking if this is an unexpected error we might actually want the server to die (after sending an error response to the API). Also, if we don't re-raise the error then we don't get a traceback in the FlowMachine logs (although I would imagine there are other ways around this).

Reply:
And I would imagine that if we're handling a generic exception that hasn't been caught by any of the more specific exception handlers then it's probably a 500 error (we should be able to predict what 400 errors a user could trigger, and handle those appropriately).

Reply:
Mm. I'm down with returning a 500, just not without sanitising content. In this case, can we return a nondescript error and add a todo to disambiguate/properly code errors from flowapi?

As to the traceback, how about using the `__traceback__` attribute of the exception object?

Reply:
```python
import traceback
try:
    assert 1 == 2
except Exception as exc:
    trace = traceback.format_list(traceback.extract_tb(exc.__traceback__))
```